### PR TITLE
fix(core): unexpected editable state after trash page refresh

### PR DESCRIPTION
**Pull request by JimmFly**
close TOV-772

### CODE REVIEW

**Comment by JimmFly** [comment]
* **#6433** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/6433?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @JimmFly and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/core/src/pages/workspace/detail-page/detail-page.tsx [hunk 1]
```diff
@@ -318,6 +321,22 @@ export const DetailPage = ({ pageId }: { pageId: string }): ReactElement => {
...
+          currentWorkspace.docCollection.awarenessStore.setReadonly(
+            page.blockSuiteDoc,
+            true
+          );
+        }
+      })
+      .catch(err => {
+        console.error(err);
+      });
+  });
```
Comment (EYHN) [inline packages/frontend/core/src/pages/workspace/detail-page/detail-page.tsx]:
```suggestion
  }, [page, isInTrash]);
```

File: packages/frontend/core/src/pages/workspace/detail-page/detail-page.tsx [hunk 2]
```diff
@@ -318,6 +321,22 @@ export const DetailPage = ({ pageId }: { pageId: string }): ReactElement => {
     }
   }, [jumpOnce, pageRecord]);
 
+  useEffect(() => {
+    currentWorkspace.engine
+      .waitForRootDocReady()
```
Comment (EYHN) [inline packages/frontend/core/src/pages/workspace/detail-page/detail-page.tsx]: Why you need to wait here?

File: packages/frontend/core/src/pages/workspace/detail-page/detail-page.tsx [hunk 3]
```diff
@@ -318,6 +321,22 @@ export const DetailPage = ({ pageId }: { pageId: string }): ReactElement => {
     }
   }, [jumpOnce, pageRecord]);
 
+  useEffect(() => {
```
Comment (pengx17) [inline packages/frontend/core/src/pages/workspace/detail-page/detail-page.tsx:319]: it seems to me this should be implemented with the new `effect` function (Livedata)

Comment (EYHN) [reply packages/frontend/core/src/pages/workspace/detail-page/detail-page.tsx:319]: No, the new `effect` is used to describe network requests 😂

**Review by EYHN — APPROVED** [review]

**Pull request merged**